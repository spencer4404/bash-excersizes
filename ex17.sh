#!/bin/bash

# Filename: ex17.sh
#
# Problem: Using awk add two more columns to the table. Calculate the change in currency 
# from the table. Then calculate the percentage of the change relative to the low value.
 
cat ex16.input | awk '{
		printf $0" "
		if ( $2 ~ /^[0-9]+/ ) {
			change = $3 - $2
			percent = (change / $2) * 100
			printf "%.4f %.4f\n", change, percent
		} else if ( $2 ~ /^[A-z]+/ ) {
			print "Variation VarPercent"
		} else {
			print ""
		}
	}'
