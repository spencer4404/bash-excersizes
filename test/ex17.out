10/06/2011 
EX_RATE	LOW	HIGH Variation VarPercent
GBP/EUR 1.1500 1.1710 0.0210 1.8261
GBP/USD 1.5330 1.5520 0.0190 1.2394
GBP/AUD 1.5720 1.6180 0.0460 2.9262
EUR/USD	1.3210 1.3440 0.0230 1.7411
 
